### ISSUE TITLE
storybook@6 throws a core-js error

### ISSUE THREAD

**Issue by 43081j** [storybookjs/storybook#11255]
With storybook 6, my _very basic_ typescript setup throws errors about core-js polyfills every time.

```
Module not found: Error: Can't resolve 'core-js/modules/web.dom-collections.iterator'
```

this is thrown from 'generated-stories-entry'. i can see earlier in the webpack output that the polyfill was built:

```
[./node_modules/@storybook/core/dist/server/common/polyfills.js] 120 bytes {vendors~main} [built]
```

which afaik is where you include core-js.

with storybook 5, this works just fine. its a clean setup from today, `npm i -D @storybook/html` (5 or 6) and this config:

```js
module.exports = {
  stories: ['../src/**/*.stories.ts'],
  webpackFinal: (config) => {
    config.module.rules.push({
      test: /\.ts$/,
      use: [
        {loader: require.resolve('ts-loader')}
      ]
    });
    config.resolve.extensions.push('.ts');
    return config;
  }
};
```

I _think_ this is because there are multiple core-js versions in my node_modules and storybook is resolving the wrong one (somehow). if i look at the one inside storybook's packages, its got the right files.

this isn't something i should have to 'fix' though IMO, as so many dependencies use core-js it seems like a very likely situation for many to bump into. my guess is you can fix it by installing core-js as a direct dependency but thats also far from ideal.

EDIT:

### Temporary workaround

If you've come across this issue while searching, we do not yet have a solution merged in (this issue will close when we do). However, there is a workaround: **simply install core-js@3 as a direct dependency** (`npm i -D core-js@3`).

If that doesn't work for you, please do comment so we are aware in case there are some other edge cases.

**Comment by shilman** [comment]
cc @ndelangen

**Comment by gerdy** [comment]
I have the same problem.
I set a breakpoint on the webpackFinal method, but it did not trigger.
I suspect that the change here caused this problem, which was good before 6.0.0 beta14.

**Comment by BWrites** [comment]
I've run into this same issue when trying to migrate from v5.3.

My webpack.config.js:
```js
const path = require('path');

module.exports = ({ config }) => {
    config.module.rules.push({
        test: /\.stories\.js?$/,
        loaders: [require.resolve('@storybook/source-loader')],
        enforce: 'pre',
    });

    config.module.rules = config.module.rules.map(rule => {
        if (
            String(rule.test) ===
            String(
                /\.(svg|ico|jpg|jpeg|png|gif|eot|otf|webp|ttf|woff|woff2|cur|ani|pdf)(\?.*)?$/
            )
        ) {
            return {
                ...rule,
                test: /\.(ico|jpg|jpeg|png|gif|eot|otf|webp|ttf|woff|woff2|cur|ani)(\?.*)?$/,
            };
        }

        return rule;
    });

    config.module.rules.push({
        test: /\.svg$/,
        use: [
            {
                loader: '@svgr/webpack',
                options: {
                    svgo: false,
                },
            },
        ],
    });

    return config;
};
```

When trying to run `build-storybook -s public -o ./docs`, I'm getting the following error in the output:
```
info => Compiling preview..
70% building 225/248 modules 23 active <LOCAL_PROJECT_PATH>/node_modules/@storybook/components/dist/Button/Button.jsERR! => Failed to build the preview
ERR! Module not found: Error: Can't resolve 'chromatic/isChromatic' in '<LOCAL_PROJECT_PATH>/node_modules/@storybook/ui/dist/components/layout'
(node:11596) UnhandledPromiseRejectionWarning: ModuleNotFoundError: Module not found: Error: Can't resolve 'chromatic/isChromatic' in '<LOCAL_PROJECT_PATH>/node_modules/@storybook/ui/dist/components/layout'
    at <LOCAL_PROJECT_PATH>/node_modules/webpack/lib/Compilation.js:925:10
    at <LOCAL_PROJECT_PATH>/node_modules/webpack/lib/NormalModuleFactory.js:401:22
    at <LOCAL_PROJECT_PATH>/node_modules/webpack/lib/NormalModuleFactory.js:130:21
    at <LOCAL_PROJECT_PATH>/node_modules/webpack/lib/NormalModuleFactory.js:224:22
    at <LOCAL_PROJECT_PATH>/node_modules/neo-async/async.js:2830:7
    at <LOCAL_PROJECT_PATH>/node_modules/neo-async/async.js:6877:13
    at <LOCAL_PROJECT_PATH>/node_modules/webpack/lib/NormalModuleFactory.js:214:25
    at <LOCAL_PROJECT_PATH>/node_modules/enhanced-resolve/lib/Resolver.js:213:14
    at <LOCAL_PROJECT_PATH>/node_modules/enhanced-resolve/lib/Resolver.js:285:5
    at eval (eval at create (<LOCAL_PROJECT_PATH>/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:13:1)
    at <LOCAL_PROJECT_PATH>/node_modules/enhanced-resolve/lib/UnsafeCachePlugin.js:44:7
    at <LOCAL_PROJECT_PATH>/node_modules/enhanced-resolve/lib/Resolver.js:285:5
    at eval (eval at create (<LOCAL_PROJECT_PATH>/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:13:1)
    at <LOCAL_PROJECT_PATH>/node_modules/enhanced-resolve/lib/Resolver.js:285:5
    at eval (eval at create (<LOCAL_PROJECT_PATH>/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:25:1)
    at <LOCAL_PROJECT_PATH>/node_modules/enhanced-resolve/lib/DescriptionFilePlugin.js:67:43
(node:11596) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:11596) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
70% building 932/932 modules 0 activenpm ERR! code ELIFECYCLE
```
*(actual project path was replaced by `<LOCAL_PROJECT_PATH>` by me)

**Comment by ndelangen** [comment]
> ERR! Module not found: Error: Can't resolve 'chromatic/isChromatic' in '<LOCAL_PROJECT_PATH>/node_modules/@storybook/ui/dist/components/layout'
(node:11596) UnhandledPromiseRejectionWarning: ModuleNotFoundError: Module not found: Error: Can't resolve 'chromatic/isChromatic' in '<LOCAL_PROJECT_PATH>/node_modules/@storybook/ui/dist/components/layout'

whut? I'm really confused.. the preview is bundling code for the manager.. and it's missing a devDependency..This makes very little sense..

**Comment by BWrites** [comment]
@ndelangen Sorry, please ignore that last error - looks like that was similar to this: #11356.  I had my `main.js` config set to

```js
module.exports = {
    stories: ['../**/*.stories.@(js)'],
    addons: [
        '@storybook/addon-a11y',
        '@storybook/addon-docs',
        '@storybook/addon-links',
        '@storybook/addon-storysource',
        '@storybook/addon-toolbars',
        '@storybook/addon-viewport',
    ],
};
```

so the glob meant it may have been loading stories from my `node_modules`. 🤦 

I've updated the glob to look solely at my `src` folder and now I'm getting the `core-js` issue mentioned by the OP...

```
info => Compiling preview..
70% building 26/29 modules 3 active <LOCAL_PROJECT_PATH>/node_modules/global/window.jsERR! => Failed to build the preview
ERR! Module not found: Error: Can't resolve 'core-js/modules/es.symbol' in '<LOCAL_PROJECT_PATH>/.storybook'
(node:3625) UnhandledPromiseRejectionWarning: ModuleNotFoundError: Module not found: Error: Can't resolve 'core-js/modules/es.symbol' in '<LOCAL_PROJECT_PATH>/.storybook'
    at <LOCAL_PROJECT_PATH>/node_modules/webpack/lib/Compilation.js:925:10
    at <LOCAL_PROJECT_PATH>/node_modules/webpack/lib/NormalModuleFactory.js:401:22
    at <LOCAL_PROJECT_PATH>/node_modules/webpack/lib/NormalModuleFactory.js:130:21
    at <LOCAL_PROJECT_PATH>/node_modules/webpack/lib/NormalModuleFactory.js:224:22
    at <LOCAL_PROJECT_PATH>/node_modules/neo-async/async.js:2830:7
    at <LOCAL_PROJECT_PATH>/node_modules/neo-async/async.js:6877:13
    at <LOCAL_PROJECT_PATH>/node_modules/webpack/lib/NormalModuleFactory.js:214:25
    at <LOCAL_PROJECT_PATH>/node_modules/enhanced-resolve/lib/Resolver.js:213:14
    at <LOCAL_PROJECT_PATH>/node_modules/enhanced-resolve/lib/Resolver.js:285:5
    at eval (eval at create (<LOCAL_PROJECT_PATH>/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:13:1)
    at <LOCAL_PROJECT_PATH>/node_modules/enhanced-resolve/lib/UnsafeCachePlugin.js:44:7
    at <LOCAL_PROJECT_PATH>/node_modules/enhanced-resolve/lib/Resolver.js:285:5
    at eval (eval at create (<LOCAL_PROJECT_PATH>/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:13:1)
    at <LOCAL_PROJECT_PATH>/node_modules/enhanced-resolve/lib/Resolver.js:285:5
    at eval (eval at create (<LOCAL_PROJECT_PATH>/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:25:1)
    at <LOCAL_PROJECT_PATH>/node_modules/enhanced-resolve/lib/DescriptionFilePlugin.js:67:43
(node:3625) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:3625) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
70% building 1953/1953 modules 0 activenpm ERR! code ELIFECYCLE
```

**Comment by BWrites** [comment]
And as @43081j suggests, adding `core-js` v3 as a dev dependency fixes it but that doesn't seem like a great solution...

**Comment by kevinkuebler** [comment]
I just ran into the same issue trying to update my storybook from 5.2.5 to 6.0.0-rc.9. Like @BWrites I added the latest `core-js` as a dev dependency and that fixed it, but I agree that it's not a great solution.

**Comment by 43081j** [comment]
just a reminder that now that you published 6.0 officially, this is still a problem.

it is very awkward having to install `core-js` directly in each repo we use storybook for. do you have any suggested fix on your end for this? anything we can contribute to help?

**Comment by shilman** [comment]
@43081j I don't think we understand the problem. Perhaps @ndelangen can share some of his wisdom here.

**Comment by 43081j** [comment]
essentially, storybook's generated output tries to require `core-js` for various polyfills.

im guessing because its generated output and doesn't "live" in the storybook directory (`node_modules/@storybook`), it will resolve the first available core-js from the top level rather than resolving it relative to storybook's directory.

so if you have storybook in a project which already has a different version of core-js (not necessarily as a direct dependency, all sorts of other projects also depend on core-js), that one may be required by accident rather than the one storybook wants.

this means the paths fail to resolve if the two core-js versions are drastically different.

**Comment by shilman** [comment]
@43081j would you be available to pair with @ndelangen on this?

**Comment by MattFisher** [comment]
Also having this problem.

Based on https://github.com/storybookjs/storybook/issues/6204#issuecomment-478992364, I added this to my config and it's resolving correctly now:
```
resolve: {
    alias: {
        'core-js/modules': path.resolve(
            __dirname,
            '..',
            'node_modules/core-js/modules'
        ),
    }
}
```
Note that this is pointing to the root `node_modules/core-js` folder in my project, not to one that's bundled with anything else.

**Comment by 43081j** [comment]
@shilman ill see if i get some free time at the weekend and will try debug it a bit at least. it'll be to do with the webpack config i guess and where it resolves modules from

my guess is we effectively run webpack from the project directory. then when we generate our webpack entry, somewhere down the tree we must require core-js. but by that point, webpack's CWD is the project dir rather than storybook's dir, so it will pick up whatever the root core-js version is instead of the one storybook wants.

**Comment by mrtwinkler** [comment]
Same here. Just set up a clean project with vue-cli, added storybook and storybook-docs. `npm run storybook` results in lots of `Module not found: Error: Can't resolve 'core-js/modules/...`.

**Comment by pmunin** [comment]
same here. Updating storybook from 5.3. to 6.0.12 - lead to bunch of errors like `Can't resolve 'core-js/modules/web.dom-collections.iterator'`

**Comment by kurt-unleashed** [comment]
Same here, even with the workaround :(

**Comment by awkale** [comment]
> Also having this problem.
> 
> Based on [#6204 (comment)](https://github.com/storybookjs/storybook/issues/6204#issuecomment-478992364), I added this to my config and it's resolving correctly now:
> 
> ```
> resolve: {
>     alias: {
>         'core-js/modules': path.resolve(
>             __dirname,
>             '..',
>             'node_modules/core-js/modules'
>         ),
>     }
> }
> ```
> 
> Note that this is pointing to the root `node_modules/core-js` folder in my project, not to one that's bundled with anything else.

@MattFisher which config file?

**Comment by mayoyml** [comment]
@awkale `main.js` :)

**Comment by 43081j** [comment]
i think the ultimate solution is that we setup the webpack config to map to the right core-js when we produce a build. but im not entirely sure where that currently happens as there's a lot of dynamic magic going on, generation of bundles etc.

if someone can point me in the right direction im happy to go figure it out and pr it.

**Comment by shilman** [comment]
@ndelangen please work with @43081j on this 🙏

**Comment by ndelangen** [comment]
I originally wrote https://github.com/ndelangen/corejs-upgrade-webpack-plugin to 'solve' this issue.

But I deprecated it, because it just adds other problems, see:
https://github.com/storybookjs/storybook/issues/7445 and https://github.com/ndelangen/corejs-upgrade-webpack-plugin/issues/7

conflicting versions of corejs are the bane of my existence.

**Comment by ndelangen** [comment]
@43081j happy to chat about a solution to this, are you on the storybook discord?
https://discord.gg/7yKH3EU

**Comment by 43081j** [comment]
no worries ill get on discord soon, wasn't aware there was a storybook one :+1:

**Comment by kelly-tock** [comment]
similar issue, I already had core-js in my project, and upgrading to 3(which we've been meaning to do anyway) doesn't seem to fix this issue.

**Comment by kelly-tock** [comment]
is it just me, if I look at core-js repo, `web.dom.iterable` doesn't exist ATM?

**Comment by 43081j** [comment]
@kelly-tock try match the exact version storybook depends on and see if that helps.

i think i found the issue now so we're just figuring out a fix, needs some discussion to make sure it doesn't happen again with another dependency.

**Comment by kelly-tock** [comment]
tried pinning to 3.0.1, no luck, the filed doesn't seem to exist. inside of the installed storybook modules, core-js ends up being installed as 3.6.5, which also doesn't have this file. gonna follow this other thread for now. can't really upgrade.

**Comment by jefferscn** [comment]
```
 "webpackFinal": (config)=> {
    config.resolve.alias['core-js/modules'] =  path.resolve(
            __dirname,
            '..',
            'node_modules/@storybook/core/node_modules/core-js/modules'
        );
    return config;
  }
```
Adding code upside to main.js sovle my problem.

**Comment by Sharcoux** [comment]
I can confirm the solution of @jefferscn . I simplified with this, but it is the same:

`config.resolve.alias['core-js/modules'] = '@storybook/core/node_modules/core-js/modules'`

**Comment by codebymikey** [comment]
You should check your `package.json`  or run `npm ls core-js` to see which version of `core-js` you're running.

If your `package.json` references `core-js` [less than `3.0.1`](https://github.com/storybookjs/storybook/commit/569b55379f5447a08b56c3e1e3652aab5f5e17a4), then you probably need to remove it, and rerun your `npm install`.

**Comment by j0hnm4r5** [comment]
I was able to fix this by adding

```
"resolutions": {
    "**/core-js": "^3.6.5"
}
```

to my `package.json`, as described in #8267.

---

Coincidentally, I was running into this problem on 6, downgraded to 5.3, hit another version of this issue after a while, implemented the fix, then upgraded and it still worked. 

I'm using Gatsby, and evidently there is one core-js@^2.4.0 dependency buried deep inside:

`npm ls core-js`
```
└─┬ gatsby@2.24.58
  ├── core-js@3.6.5  deduped
  └─┬ gatsby-cli@2.12.96
    └─┬ yurnalist@1.1.2
      └─┬ babel-runtime@6.26.0
        └── UNMET DEPENDENCY core-js@^2.4.0
```

Adding the `resolutions` field does technically break the core-js dependency for that Gatsby / babel-runtime dependency, but I haven't seen an issue yet.

**Comment by Sharcoux** [comment]
babel-jest 26.2.2 is using babel-runtime 6.26.0 which requires core-js 2.4.0.

**Comment by shraddha-chadha** [comment]
I had the same issue with stroybook v6 + React + Typescript and this solved my issue

.storybook/main.js
```
const path = require('path');

module.exports = {
  stories: ['../src/**/*.stories.@(ts|tsx)'],
  addons: ['@storybook/addon-actions', '@storybook/addon-links', '@storybook/addon-docs'],
  webpackFinal: (config) => {
    config.module.rules.push({
      test: /\.(ts|tsx)$/,
      use: [
        {
          loader: 'babel-loader',
          options: {
            presets: ['babel-preset-react-app']
          }
        }
      ],
      include: path.resolve(__dirname, '../src'),
      exclude: /node_modules/,
    });
    return config;
  },
};
```

And .storybook/.babelrc
`{
  "presets": [
    "babel-preset-react-app"
  ]
}`

**Comment by tomasklingen** [comment]
Problem for me was that I didn't match the `@babel/preset-env` option `corejs` property version to the correct one.

package.json:
`		"core-js": "^3.6.5",`

.babelrc (or wherever your babel config lives)
```
		[
			"@babel/preset-env",
			{
				"corejs": "3.6.5", <- make sure this version is the same as core-js version installed!
				...
			}
		],
```

**Comment by sumcoding** [comment]
Hey, also running into this issue as well. I have a mono repo and import a lot off assets from a shared folder outside of the main storybook app's. I had to install core-js within that shared folder, not ideal (despite having it installed in the storybook folder) and now I am running into more issues. 

A lot of my documentation is shared as well, so need to import `mdx` docs from there into the storybook app, this is resulting in errors as well because storybook seems to think that the shared folder should be where it imports resources from as it relates to those docs.

`Error: Can't resolve '@storybook/addon-docs/blocks'` (one of many)


main.js 
```
stories: [
    '../stories/**/*.stories.mdx',
    '../stories/**/*.stories.@(js|mdx)',
    path.resolve(__dirname, '../../shared/.storybook/documentation/**/*.stories.mdx')
  ],
```

**Comment by hydrosquall** [comment]
> I can confirm the solution of @jefferscn . I simplified with this, but it is the same:
> 
> `config.resolve.alias['core-js/modules'] = '@storybook/core/node_modules/core-js/modules'`

Unfortunately this didn't work for me in a yarn 2 (berry) project, since the `/node_modules/` folder is not directly accessible. My main application already specifies `core-js` as an explicit dependency

The error I looks like this:

```bash
 Error: Your application tried to access core-js, but it isn't declared in your dependencies; this makes the require call ambiguous and unsound.
```

This issue may be related (but I'm not sure).

https://github.com/yarnpkg/berry/issues/947#issue-564792093

**Comment by kevinhaas** [comment]
> 
> 
> ```
>  "webpackFinal": (config)=> {
>     config.resolve.alias['core-js/modules'] =  path.resolve(
>             __dirname,
>             '..',
>             'node_modules/@storybook/core/node_modules/core-js/modules'
>         );
>     return config;
>   }
> ```
> 
> Adding code upside to main.js sovle my problem.

I had this fix in and it was working, but I upgraded all of my packages today and I ended up getting core-js errors again when trying to start storybook. I noticed that this path to storybooks core-js dep no longer exists.

[I added core-js and removed the above fix](https://github.com/storybookjs/storybook/issues/11633#issuecomment-687933005), and everything works again.

I didn't look into it much, but one thing I noticed was previously I had 6.0.21, which had this in the package.json:
```
  "_phantomChildren": {
    "p-limit": "2.3.0",
    "picomatch": "2.2.2"
  },
```

6.0.26 contains:
```
  "_phantomChildren": {
    "core-js": "3.6.5",
    "p-limit": "2.3.0",
    "picomatch": "2.2.2"
  },
```

**Comment by jefferscn** [comment]
Because your project use the same version core-js with storybooks.

**Comment by jefferscn** [comment]
> > ```
> >  "webpackFinal": (config)=> {
> >     config.resolve.alias['core-js/modules'] =  path.resolve(
> >             __dirname,
> >             '..',
> >             'node_modules/@storybook/core/node_modules/core-js/modules'
> >         );
> >     return config;
> >   }
> > ```
> > 
> > 
> > Adding code upside to main.js sovle my problem.
> 
> I had this fix in and it was working, but I upgraded all of my packages today and I ended up getting core-js errors again when trying to start storybook. I removed the above fix, and everything works again.
> 
> I didn't look into it much, but one thing I noticed was previously I had 6.0.21, which had this in the package.json:
> 
> ```
>   "_phantomChildren": {
>     "p-limit": "2.3.0",
>     "picomatch": "2.2.2"
>   },
> ```
> 
> 6.0.26 contains:
> 
> ```
>   "_phantomChildren": {
>     "core-js": "3.6.5",
>     "p-limit": "2.3.0",
>     "picomatch": "2.2.2"
>   },
> ```
> 
> I'm not sure how exactly things are resolving here without doing some reading, but something definitely happened with how core-js is resolved. Not sure if this is project specific, though.

Because your project now has the same version core-js with storybooks.

**Comment by timgivois** [comment]
I had to modify my `.storybook/main.js` file to:
```
    webpackFinal: async config => {
        return {
            ...config,
            resolve: {
                alias: {
                    'core-js/modules': '@storybook/core/node_modules/core-js/modules',
                    'core-js/features': '@storybook/core/node_modules/core-js/features',
                },
            },
        };
```

**Comment by dejanvasic85** [comment]
Here's my case. As soon as bringing in `graphql-tools` dependency for mocking GraphQL data ([based on this neat approach from stripe](https://www.freecodecamp.org/news/a-new-approach-to-mocking-graphql-data-1ef49de3d491/)), it brings up this error.

All I have to do is add this bit of code in any Story (or preview.js):

```
import {
  makeExecutableSchema,
  addMockFunctionsToSchema
} from 'graphql-tools';

const schemaString = `
  type Todo { id: ID, text: String, completed: Boolean }
  type User { id: ID, name: String }
  type Query { todoItems: [Todo] }
  # ... other types here
`;

// Make a GraphQL schema with no resolvers
const schema = makeExecutableSchema({ typeDefs: schemaString });
```

Bunch of errors appear starting with:
```
ERROR in ./node_modules/@graphql-tools/git-loader/index.esm.js
Module not found: Error: Can't resolve 'child_process' in 'xxx/node_modules/@graphql-tools/git-loader'

ERROR in ./node_modules/@graphql-tools/load-files/node_modules/@nodelib/fs.stat/out/adapters/fs.js
Module not found: Error: Can't resolve 'fs' in '/xxx/node_modules/@graphql-tools/load-files/node_modules/@nodelib/fs.stat/out/adapters'

ERROR in ./node_modules/@graphql-tools/load-files/node_modules/fast-glob/out/settings.js
Module not found: Error: Can't resolve 'fs' in '/xxx/node_modules/@graphql-tools/load-files/node_modules/fast-glob/out'

...
```

Running the command `npm ls core-js`. The only output that contains core-js < v3 is this:
```
└─┬ graphql-tools@7.0.1
  └─┬ @graphql-tools/relay-operation-optimizer@6.2.5
    └─┬ relay-compiler@10.0.1
      └─┬ fbjs@1.0.0
        └── core-js@2.6.11
```

Neither solutions worked for me. The resolution was not successful in package.json :( 

```
"resolutions": {
    "styled-components": "^5",
    "**/core-js": "^3.6.5"
  }
```

**Comment by shilman** [comment]
@dejanvasic85 Did you try the `--no-dll` flag?

**Comment by epiqueras** [comment]
I did @shilman.

I also tried all the proposed solutions in this issue with Next.js, and none of them worked.

This is making v6 unusable.

**Comment by shilman** [comment]
@epiqueras do you have a repro repo you can share?

**Comment by epiqueras** [comment]
I am installing it in a new Next.js project.

**Comment by kevinmpowell** [comment]
@timgivois workaround worked for me: https://github.com/storybookjs/storybook/issues/11255#issuecomment-713083086

I'm using the Web Components flavor of Storybook.

**Comment by shilman** [comment]
Please try upgrading to 6.1

```
npx sb upgrade --prerelease
```

**Comment by epiqueras** [comment]
Still broken.

**Comment by barryofguilder** [comment]
It's still broken for me too and this is with a brand new Ember.js app that has nothing in it yet except for Storybook.

**Comment by epiqueras** [comment]
The issue arises when any dependency you have uses `core-js@<3`.

It seems that Storybook forces all dependencies to use the same version.

**Comment by FilipMessa** [comment]
I installed  "core-js": "^3.6.5", as devDependency and this solve this issue for me.

**Comment by ktrzos** [comment]
I'm having this issue also.

> Module not found: Error: Can't resolve 'core-js/modules/web.dom.iterable'

The interesting part is, that it occurs while having this in `.babelrc`:

```
["@babel/plugin-proposal-class-properties", { "loose": true }],
["@babel/plugin-proposal-private-methods", { "loose": true }]
```

When I'm removing `{ "loose": true }` from the 2nd one,

```
["@babel/plugin-proposal-class-properties", { "loose": true }],
["@babel/plugin-proposal-private-methods"]
```


I got only a warning

> Module build failed [...] 'loose' mode configuration must be the same for @babel/plugin-proposal-class-properties, @babel/plugin-proposal-private-methods and @babel/plugin-proposal-private-property-in-object (when they are enabled).

Using Storybook 6.1.0-beta.3.

**Comment by 43081j** [comment]
For people still bumping into this, we're blocked by babel/babel#12035

The problem is pretty well understood now and is babel's fault. Workarounds may be possible but would be fairly complex and hacky so im hoping babel can just fix it on their end instead, or at least tell us how to.

**Comment by mayashavin** [comment]
I have a Vue component library that has zero dependencies needed for `core-js`, and it still throw me error
`Module not found: Error: Can't resolve 'core-js/modules/web.dom.iterable'`
After spending like >1 hour trying to install `babel`, `core-js`, and tried all the workaround, it still gave me error 🤦

**Comment by shilman** [comment]
Zoinks!! I just released https://github.com/storybookjs/storybook/releases/tag/v6.2.0-alpha.0 containing PR #13055 that references this issue. Upgrade today to the `@next` NPM tag to try it out!

```
npx sb upgrade --prerelease
```

Closing this issue. Please re-open if you think there's still more to do.

**Comment by kelly-tock** [comment]
Can this be updated when it's fully released?

**Comment by barryofguilder** [comment]
@shilman I just upgraded my Ember Addon with the pre-release version and it all works!

**Comment by mayashavin** [comment]
@shilman Still not working for me for very weird reason. Running `yarn storybook` throws a bunch of errors about core-js polyfill. For example:

```
ERROR in ./.storybook/preview.js-generated-config-entry.js
Module not found: Error: Can't resolve 'core-js/modules/es.symbol' in ...
 @ ./.storybook/preview.js-generated-config-entry.js 3:0-36
 @ multi ./node_modules/@storybook/vue/node_modules/@storybook/core/dist/server/common/polyfills.js ./node_modules/@storybook/vue/node_modules/@storybook/core/dist/server/preview/globals.js ./.storybook/storybook-init-framework-entry.js ./node_modules/@storybook/addon-docs/dist/frameworks/common/config.js-generated-other-entry.js ./node_modules/@storybook/addon-docs/dist/frameworks/vue/config.js-generated-other-entry.js ./node_modules/@storybook/addon-links/dist/preset/addDecorator.js-generated-other-entry.js ./node_modules/@storybook/addon-actions/dist/preset/addDecorator.js-generated-other-entry.js ./node_modules/@storybook/addon-actions/dist/preset/addArgs.js-generated-other-entry.js ./node_modules/@storybook/addon-backgrounds/dist/preset/addDecorator.js-generated-other-entry.js ./node_modules/@storybook/addon-backgrounds/dist/preset/addParameter.js-generated-other-entry.js ./.storybook/preview.js-generated-config-entry.js ./.storybook/generated-stories-entry.js (webpack)-hot-middleware/client.js?reload=true&quiet=false&noInfo=undefined
```
I have to do the workaround in `main.js` with the following:
```
webpackFinal: (config) => {
    config.resolve.alias["core-js/modules"] =
      "@storybook/core/node_modules/core-js/modules";
    config.resolve.alias["core-js/features"] =
      "@storybook/core/node_modules/core-js/features";
    return config;
  },
```
This happens when I have `core-js` version specifically points to 2.6.9.

**Comment by madroneropaulo** [comment]
Still doesn't work. Just updated with the patch 6.2.0-alpha.1

**Comment by ben-reitz** [comment]
This release has finally fixed it for our component library's Storybook - thanks!

**Comment by shilman** [comment]
For those of you whose projects are NOT fixed by `6.2.0-alpha.1`, some suggestions:
- Try removing `node_modules` and/or lockfiles and reinstalling?
- Try removing whatever workarounds you've added before

If that still doesn't fix it, perhaps you can post a link to your repo and we might be able to take a look and see what's going on. Thanks and thanks for your patience on this!

**Comment by hydrosquall** [comment]
Hey, thanks for the continued work on this. Unfortunately, upgrading to `6.2.0-alpha.1` on a private repo didn't fix the stories, we didn't have any workarounds to remove. I'll update here if I come up with a minimal repo that contains the issue.

The repository is a mix of JS/typescript, managed by yarn 2 (berry), and has core-js version `3.0.1` specified at the root level `package.json`. Most of the errors are of the form

```bash
Module not found: Error: Your application tried to access core-js, but it isn't declared in your dependencies; this makes the require call ambiguous and unsound.

Required package: core-js (via "core-js/modules/es.string.iterator")
// path to some internal code
```

**Comment by henriquez** [comment]
I had this problem as well, to fix it I had to delete my node_modules and package-lock.json and update storybook and its add ons to alpha.6, including
"@storybook/addon-actions": "^6.2.0-alpha.6",
"@storybook/addon-links": "^6.2.0-alpha.6",
 "@storybook/addons": "^6.2.0-alpha.6",
"@storybook/svelte": "^6.2.0-alpha.6",

**Comment by uriklar** [comment]
Hi @shilman, I'm seeing this error using v6.2.0-alpha.10.

My use case is that we have a library in node_modules that we are bundeling in our app, so I added this to the custom webpack config:

<details>
  <summary>Custom webpack config</summary>

```
module.exports = {
    webpackFinal: async (config, { configType }) => {
        // `configType` has a value of 'DEVELOPMENT' or 'PRODUCTION'
        // You can change the configuration based on that.
        // 'PRODUCTION' is used when building the static version of storybook.

        // Make whatever fine-grained changes you need

        const { exclude, ...duplicatedConfig } = config.module.rules[0];

        config.module.rules[0] = {
            ...duplicatedConfig,
            include: [
                ...duplicatedConfig.include,
                fs.realpathSync(`${__dirname}/../node_modules/@honeybook/hbui`),
                fs.realpathSync(`${__dirname}/../node_modules/@honeybook/icons`)
            ]
        };

        // Return the altered config
        return config;
    }
};
```
  </details>



<details>
  <summary>Runtime error:</summary>
  
```
export.js:2 Uncaught TypeError: __webpack_require__(...) is not a function
    at Object../node_modules/core-js/internals/export.js (export.js:2)
    at __webpack_require__ (bootstrap:848)
    at fn (bootstrap:150)
    at Object../node_modules/core-js/modules/es.object.define-property.js (es.object.define-property.js:1)
    at __webpack_require__ (bootstrap:848)
    at fn (bootstrap:150)
    at Object../node_modules/webpack/buildin/module.js (harmony-module.js:24)
    at __webpack_require__ (bootstrap:848)
    at fn (bootstrap:150)
    at Object../node_modules/regenerator-runtime/runtime.js (runtime.js:8)
```
 </details>
 
<details>
   <summary>Webpack debug output:</summary>
   
   ```
   {
  mode: 'development',
  bail: false,
  devtool: '#cheap-module-source-map',
  entry: [
    '/Users/uriklar/Code/honeybook-react/node_modules/@storybook/core/dist/cjs/server/common/polyfills.js',
    '/Users/uriklar/Code/honeybook-react/node_modules/@storybook/core/dist/cjs/server/preview/globals.js',
    '/Users/uriklar/Code/honeybook-react/.storybook/storybook-init-framework-entry.js',
    '/Users/uriklar/Code/honeybook-react/.storybook/preview.js-generated-config-entry.js',
    '/Users/uriklar/Code/honeybook-react/node_modules/webpack-hot-middleware/client.js?reload=true&quiet=false&noInfo=undefined'
  ],
  output: {
    path: '/Users/uriklar/Code/honeybook-react/node_modules/.cache/storybook/public',
    filename: '[name].[hash].bundle.js',
    publicPath: ''
  },
  plugins: [
    FilterWarningsPlugin {
      exclude: [ /export '\S+' was not found in 'global'/ ]
    },
    VirtualModulesPlugin {
      _staticModules: {
        '/Users/uriklar/Code/honeybook-react/.storybook/storybook-init-framework-entry.js': "import '@storybook/react';",
        '/Users/uriklar/Code/honeybook-react/.storybook/preview.js-generated-config-entry.js': 'function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }\n' +
          '\n' +
          'function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }\n' +
          '\n' +
          'function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }\n' +
          '\n' +
          '/* eslint-disable import/no-unresolved */\n' +
          "import { addDecorator, addParameters, addLoader, addArgTypesEnhancer } from '/Users/uriklar/Code/honeybook-react/node_modules/@storybook/client-api';\n" +
          "import { logger } from '/Users/uriklar/Code/honeybook-react/node_modules/@storybook/client-logger';\n" +
          "import * as config from '/Users/uriklar/Code/honeybook-react/.storybook/preview.js';\n" +
          'Object.keys(config).forEach(function (key) {\n' +
          '  var value = config[key];\n' +
          '\n' +
          '  switch (key) {\n' +
          "    case 'args':\n" +
          "    case 'argTypes':\n" +
          '      {\n' +
          "        return logger.warn('Invalid args/argTypes in config, ignoring.', JSON.stringify(value));\n" +
          '      }\n' +
          '\n' +
          "    case 'decorators':\n" +
          '      {\n' +
          '        return value.forEach(function (decorator) {\n' +
          '          return addDecorator(decorator, false);\n' +
          '        });\n' +
          '      }\n' +
          '\n' +
          "    case 'loaders':\n" +
          '      {\n' +
          '        return value.forEach(function (loader) {\n' +
          '          return addLoader(loader, false);\n' +
          '        });\n' +
          '      }\n' +
          '\n' +
          "    case 'parameters':\n" +
          '      {\n' +
          '        return addParameters(_objectSpread({}, value), false);\n' +
          '      }\n' +
          '\n' +
          "    case 'argTypesEnhancers':\n" +
          '      {\n' +
          '        return value.forEach(function (enhancer) {\n' +
          '          return addArgTypesEnhancer(enhancer);\n' +
          '        });\n' +
          '      }\n' +
          '\n' +
          "    case 'globals':\n" +
          "    case 'globalTypes':\n" +
          '      {\n' +
          '        var v = {};\n' +
          '        v[key] = value;\n' +
          '        return addParameters(v, false);\n' +
          '      }\n' +
          '\n' +
          '    default:\n' +
          '      {\n' +
          '        // eslint-disable-next-line prefer-template\n' +
          "        return console.log(key + ' was not supported :( !');\n" +
          '      }\n' +
          '  }\n' +
          '});',
        '/Users/uriklar/Code/honeybook-react/.storybook/generated-stories-entry.js': '"use strict";\n' +
          '\n' +
          'var _frameworkImportPath = require("@storybook/react");\n' +
          '\n' +
          '/* eslint-disable import/no-unresolved */\n' +
          '(0, _frameworkImportPath.configure)([], module, false);'
      }
    },
    HtmlWebpackPlugin {
      options: {
        template: '/Users/uriklar/Code/honeybook-react/node_modules/@storybook/core/dist/cjs/server/templates/index.ejs',
        templateContent: false,
        templateParameters: [Function: templateParameters],
        filename: 'iframe.html',
        publicPath: 'auto',
        hash: false,
        inject: false,
        scriptLoading: 'blocking',
        compile: true,
        favicon: false,
        minify: {
          collapseWhitespace: true,
          removeComments: true,
          removeRedundantAttributes: true,
          removeScriptTypeAttributes: false,
          removeStyleLinkTypeAttributes: true,
          useShortDoctype: true
        },
        cache: true,
        showErrors: true,
        chunks: 'all',
        excludeChunks: [],
        chunksSortMode: 'none',
        meta: {},
        base: false,
        title: 'Webpack App',
        xhtml: false,
        alwaysWriteToDisk: true
      },
      childCompilerHash: undefined,
      assetJson: undefined,
      hash: undefined,
      version: 4
    },
    DefinePlugin {
      definitions: {
        'process.env': {
          NODE_ENV: '"development"',
          NODE_PATH: '""',
          STORYBOOK: '"true"',
          PUBLIC_URL: '"."'
        },
        NODE_ENV: '"development"'
      }
    },
    WatchMissingNodeModulesPlugin {
      nodeModulesPath: '/Users/uriklar/Code/honeybook-react/node_modules'
    },
    HotModuleReplacementPlugin {
      options: {},
      multiStep: undefined,
      fullBuildTimeout: 200,
      requestTimeout: 10000
    },
    CaseSensitivePathsPlugin {
      options: {},
      logger: {
        log: [Function: bound consoleCall],
        warn: [Function: bound consoleCall],
        dir: [Function: bound consoleCall],
        time: [Function: bound consoleCall],
        timeEnd: [Function: bound consoleCall],
        timeLog: [Function: bound consoleCall],
        trace: [Function: bound consoleCall],
        assert: [Function: bound consoleCall],
        clear: [Function: bound consoleCall],
        count: [Function: bound consoleCall],
        countReset: [Function: bound consoleCall],
        group: [Function: bound consoleCall],
        groupEnd: [Function: bound consoleCall],
        table: [Function: bound consoleCall],
        debug: [Function: bound consoleCall],
        info: [Function: bound consoleCall],
        dirxml: [Function: bound consoleCall],
        error: [Function: bound consoleCall],
        groupCollapsed: [Function: bound consoleCall],
        Console: [Function: Console],
        profile: [Function: profile],
        profileEnd: [Function: profileEnd],
        timeStamp: [Function: timeStamp],
        context: [Function: context],
        [Symbol(kBindStreamsEager)]: [Function: bound ],
        [Symbol(kBindStreamsLazy)]: [Function: bound ],
        [Symbol(kBindProperties)]: [Function: bound ],
        [Symbol(kWriteToConsole)]: [Function: bound ],
        [Symbol(kGetInspectOptions)]: [Function: bound ],
        [Symbol(kFormatForStdout)]: [Function: bound ],
        [Symbol(kFormatForStderr)]: [Function: bound ]
      },
      pathCache: {},
      fsOperations: 0,
      primed: false
    },
    ProgressPlugin {
      profile: false,
      handler: undefined,
      modulesCount: 500,
      showEntries: false,
      showModules: true,
      showActiveModules: true
    },
    DefinePlugin { definitions: {} },
    DocgenPlugin {
      name: 'React Docgen Typescript Plugin',
      options: {
        shouldExtractLiteralValuesFromEnum: true,
        shouldRemoveUndefinedFromOptional: true,
        propFilter: [Function: propFilter]
      }
    }
  ],
  module: {
    rules: [
      {
        test: /\.(mjs|tsx?|jsx?)$/,
        use: [
          {
            loader: '/Users/uriklar/Code/honeybook-react/node_modules/babel-loader/lib/index.js',
            options: {
              cacheDirectory: '/Users/uriklar/Code/honeybook-react/node_modules/.cache/storybook/babel',
              sourceType: 'unambiguous',
              presets: [
                [
                  '/Users/uriklar/Code/honeybook-react/node_modules/@babel/preset-env/lib/index.js',
                  { shippedProposals: true }
                ],
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/preset-typescript/lib/index.js',
                [
                  '/Users/uriklar/Code/honeybook-react/node_modules/@babel/preset-react/lib/index.js',
                  { runtime: 'automatic' }
                ],
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/preset-flow/lib/index.js'
              ],
              plugins: [
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-shorthand-properties/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-block-scoping/lib/index.js',
                [
                  '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-decorators/lib/index.js',
                  { legacy: true }
                ],
                [
                  '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-class-properties/lib/index.js',
                  { loose: true }
                ],
                [
                  '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-private-methods/lib/index.js',
                  { loose: true }
                ],
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-export-default-from/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-syntax-dynamic-import/lib/index.js',
                [
                  '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js',
                  { loose: true, useBuiltIns: true }
                ],
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-classes/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-arrow-functions/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-parameters/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-destructuring/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-spread/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-for-of/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/babel-plugin-macros/dist/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-optional-chaining/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-nullish-coalescing-operator/lib/index.js',
                [
                  '/Users/uriklar/Code/honeybook-react/node_modules/babel-plugin-polyfill-corejs3/lib/index.js',
                  {
                    method: 'usage-global',
                    absoluteImports: '/Users/uriklar/Code/honeybook-react/node_modules/core-js/index.js',
                    version: '3.8.1'
                  }
                ],
                [
                  '/Users/uriklar/Code/honeybook-react/node_modules/babel-plugin-emotion/dist/babel-plugin-emotion.cjs.js',
                  { sourceMap: true, autoLabel: true }
                ],
                '/Users/uriklar/Code/honeybook-react/node_modules/babel-plugin-add-react-displayname/index.js'
              ],
              overrides: [
                {
                  test: /\.(mjs|jsx?)$/,
                  plugins: [
                    [
                      '/Users/uriklar/Code/honeybook-react/node_modules/babel-plugin-react-docgen/lib/index.js',
                      {
                        DOC_GEN_COLLECTION_NAME: 'STORYBOOK_REACT_CLASSES'
                      }
                    ]
                  ]
                }
              ]
            }
          }
        ],
        include: [
          '/Users/uriklar/Code/honeybook-react',
          '/Users/uriklar/Code/honeybook-react/node_modules/@honeybook/hbui',
          '/Users/uriklar/Code/honeybook-react/node_modules/@honeybook/icons'
        ]
      },
      {
        test: /\.js$/,
        use: [
          {
            loader: '/Users/uriklar/Code/honeybook-react/node_modules/babel-loader/lib/index.js',
            options: {
              sourceType: 'unambiguous',
              presets: [
                [
                  '/Users/uriklar/Code/honeybook-react/node_modules/@babel/preset-env/lib/index.js',
                  {
                    shippedProposals: true,
                    modules: false,
                    targets: 'defaults'
                  }
                ],
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/preset-react/lib/index.js'
              ],
              plugins: [
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-shorthand-properties/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-block-scoping/lib/index.js',
                [
                  '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-decorators/lib/index.js',
                  { legacy: true }
                ],
                [
                  '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-class-properties/lib/index.js',
                  { loose: true }
                ],
                [
                  '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-private-methods/lib/index.js',
                  { loose: true }
                ],
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-export-default-from/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-syntax-dynamic-import/lib/index.js',
                [
                  '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js',
                  { loose: true, useBuiltIns: true }
                ],
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-classes/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-arrow-functions/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-parameters/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-destructuring/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-spread/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-for-of/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/babel-plugin-macros/dist/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-optional-chaining/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-nullish-coalescing-operator/lib/index.js',
                [
                  '/Users/uriklar/Code/honeybook-react/node_modules/babel-plugin-polyfill-corejs3/lib/index.js',
                  {
                    method: 'usage-global',
                    absoluteImports: '/Users/uriklar/Code/honeybook-react/node_modules/core-js/index.js',
                    version: '3.8.1'
                  }
                ]
              ]
            }
          }
        ],
        include: [Function: include]
      },
      {
        test: /\.md$/,
        use: [
          {
            loader: '/Users/uriklar/Code/honeybook-react/node_modules/raw-loader/dist/cjs.js'
          }
        ]
      },
      {
        test: /\.css$/,
        sideEffects: true,
        use: [
          '/Users/uriklar/Code/honeybook-react/node_modules/style-loader/dist/cjs.js',
          {
            loader: '/Users/uriklar/Code/honeybook-react/node_modules/css-loader/dist/cjs.js',
            options: { importLoaders: 1 }
          },
          {
            loader: '/Users/uriklar/Code/honeybook-react/node_modules/postcss-loader/src/index.js',
            options: {
              ident: 'postcss',
              postcss: {},
              plugins: [Function: plugins]
            }
          }
        ]
      },
      {
        test: /\.(svg|ico|jpg|jpeg|png|apng|gif|eot|otf|webp|ttf|woff|woff2|cur|ani|pdf)(\?.*)?$/,
        loader: '/Users/uriklar/Code/honeybook-react/node_modules/file-loader/dist/cjs.js',
        options: { name: 'static/media/[name].[hash:8].[ext]', esModule: false }
      },
      {
        test: /\.(mp4|webm|wav|mp3|m4a|aac|oga)(\?.*)?$/,
        loader: '/Users/uriklar/Code/honeybook-react/node_modules/url-loader/dist/cjs.js',
        query: { limit: 10000, name: 'static/media/[name].[hash:8].[ext]' }
      }
    ]
  },
  resolve: {
    extensions: [
      '.mjs', '.js',
      '.jsx', '.ts',
      '.tsx', '.json',
      '.cjs'
    ],
    modules: [ 'node_modules' ],
    mainFields: [ 'browser', 'main' ],
    alias: {
      '@emotion/core': '/Users/uriklar/Code/honeybook-react/node_modules/@emotion/core',
      '@emotion/styled': '/Users/uriklar/Code/honeybook-react/node_modules/@emotion/styled',
      'emotion-theming': '/Users/uriklar/Code/honeybook-react/node_modules/emotion-theming',
      '@storybook/addons': '/Users/uriklar/Code/honeybook-react/node_modules/@storybook/addons',
      '@storybook/api': '/Users/uriklar/Code/honeybook-react/node_modules/@storybook/api',
      '@storybook/channels': '/Users/uriklar/Code/honeybook-react/node_modules/@storybook/channels',
      '@storybook/channel-postmessage': '/Users/uriklar/Code/honeybook-react/node_modules/@storybook/channel-postmessage',
      '@storybook/components': '/Users/uriklar/Code/honeybook-react/node_modules/@storybook/components',
      '@storybook/core-events': '/Users/uriklar/Code/honeybook-react/node_modules/@storybook/core-events',
      '@storybook/router': '/Users/uriklar/Code/honeybook-react/node_modules/@storybook/router',
      '@storybook/theming': '/Users/uriklar/Code/honeybook-react/node_modules/@storybook/theming',
      '@storybook/semver': '/Users/uriklar/Code/honeybook-react/node_modules/@storybook/core/node_modules/@storybook/semver',
      '@storybook/client-api': '/Users/uriklar/Code/honeybook-react/node_modules/@storybook/client-api',
      '@storybook/client-logger': '/Users/uriklar/Code/honeybook-react/node_modules/@storybook/client-logger',
      react: '/Users/uriklar/Code/honeybook-react/node_modules/react',
      'react-dom': '/Users/uriklar/Code/honeybook-react/node_modules/react-dom'
    },
    plugins: [
      {
        apply: [Function: nothing],
        makePlugin: [Function],
        moduleLoader: [Function],
        topLevelLoader: { apply: [Function: nothing] },
        bind: [Function],
        tsLoaderOptions: [Function],
        forkTsCheckerOptions: [Function]
      }
    ]
  },
  resolveLoader: { plugins: [ { apply: [Function: nothing] } ] },
  optimization: {
    splitChunks: { chunks: 'all' },
    runtimeChunk: true,
    sideEffects: true,
    usedExports: true,
    concatenateModules: true,
    minimizer: []
  },
  performance: { hints: false }
}
Manager webpack config
{
  name: 'manager',
  mode: 'development',
  bail: false,
  devtool: 'none',
  entry: [
    '/Users/uriklar/Code/honeybook-react/node_modules/@storybook/core/dist/cjs/server/common/polyfills.js',
    '/Users/uriklar/Code/honeybook-react/node_modules/@storybook/core/dist/cjs/client/manager/index.js',
    '/Users/uriklar/Code/honeybook-react/.storybook/generated-refs.js'
  ],
  output: {
    path: '/Users/uriklar/Code/honeybook-react/node_modules/.cache/storybook/public',
    filename: '[name].[chunkhash].bundle.js',
    publicPath: ''
  },
  cache: FileSystemCache {
    basePath: '/Users/uriklar/Code/honeybook-react/node_modules/.cache/storybook/dev-server',
    ns: '325c8f456729b912b0d2134054eb7448'
  },
  plugins: [
    VirtualModulesPlugin {
      _staticModules: {
        '/Users/uriklar/Code/honeybook-react/.storybook/generated-refs.js': '"use strict";\n' +
          '\n' +
          'var _addons = require("@storybook/addons");\n' +
          '\n' +
          '_addons.addons.setConfig({\n' +
          '  refs: {}\n' +
          '});'
      }
    },
    HtmlWebpackPlugin {
      options: {
        template: '/Users/uriklar/Code/honeybook-react/node_modules/@storybook/core/dist/cjs/server/templates/index.ejs',
        templateContent: false,
        templateParameters: [Function: templateParameters],
        filename: 'index.html',
        publicPath: 'auto',
        hash: false,
        inject: false,
        scriptLoading: 'blocking',
        compile: true,
        favicon: false,
        minify: 'auto',
        cache: true,
        showErrors: true,
        chunks: 'all',
        excludeChunks: [],
        chunksSortMode: 'none',
        meta: {},
        base: false,
        title: 'Webpack App',
        xhtml: false,
        alwaysWriteToDisk: true
      },
      childCompilerHash: undefined,
      assetJson: undefined,
      hash: undefined,
      version: 4
    },
    CaseSensitivePathsPlugin {
      options: {},
      logger: {
        log: [Function: bound consoleCall],
        warn: [Function: bound consoleCall],
        dir: [Function: bound consoleCall],
        time: [Function: bound consoleCall],
        timeEnd: [Function: bound consoleCall],
        timeLog: [Function: bound consoleCall],
        trace: [Function: bound consoleCall],
        assert: [Function: bound consoleCall],
        clear: [Function: bound consoleCall],
        count: [Function: bound consoleCall],
        countReset: [Function: bound consoleCall],
        group: [Function: bound consoleCall],
        groupEnd: [Function: bound consoleCall],
        table: [Function: bound consoleCall],
        debug: [Function: bound consoleCall],
        info: [Function: bound consoleCall],
        dirxml: [Function: bound consoleCall],
        error: [Function: bound consoleCall],
        groupCollapsed: [Function: bound consoleCall],
        Console: [Function: Console],
        profile: [Function: profile],
        profileEnd: [Function: profileEnd],
        timeStamp: [Function: timeStamp],
        context: [Function: context],
        [Symbol(kBindStreamsEager)]: [Function: bound ],
        [Symbol(kBindStreamsLazy)]: [Function: bound ],
        [Symbol(kBindProperties)]: [Function: bound ],
        [Symbol(kWriteToConsole)]: [Function: bound ],
        [Symbol(kGetInspectOptions)]: [Function: bound ],
        [Symbol(kFormatForStdout)]: [Function: bound ],
        [Symbol(kFormatForStderr)]: [Function: bound ]
      },
      pathCache: {},
      fsOperations: 0,
      primed: false
    },
    DefinePlugin { definitions: {} },
    DefinePlugin {
      definitions: {
        'process.env': {
          NODE_ENV: '"development"',
          NODE_PATH: '""',
          STORYBOOK: '"true"',
          PUBLIC_URL: '""'
        },
        NODE_ENV: '"development"'
      }
    }
  ],
  module: {
    rules: [
      {
        test: /\.(mjs|tsx?|jsx?)$/,
        use: [
          {
            loader: '/Users/uriklar/Code/honeybook-react/node_modules/babel-loader/lib/index.js',
            options: {
              sourceType: 'unambiguous',
              presets: [
                [
                  '/Users/uriklar/Code/honeybook-react/node_modules/@babel/preset-env/lib/index.js',
                  { shippedProposals: true }
                ],
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/preset-typescript/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/preset-react/lib/index.js'
              ],
              plugins: [
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-shorthand-properties/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-block-scoping/lib/index.js',
                [
                  '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-decorators/lib/index.js',
                  { legacy: true }
                ],
                [
                  '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-class-properties/lib/index.js',
                  { loose: true }
                ],
                [
                  '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-private-methods/lib/index.js',
                  { loose: true }
                ],
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-export-default-from/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-syntax-dynamic-import/lib/index.js',
                [
                  '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js',
                  { loose: true, useBuiltIns: true }
                ],
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-classes/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-arrow-functions/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-parameters/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-destructuring/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-spread/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-for-of/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/babel-plugin-macros/dist/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-optional-chaining/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-nullish-coalescing-operator/lib/index.js',
                [
                  '/Users/uriklar/Code/honeybook-react/node_modules/babel-plugin-polyfill-corejs3/lib/index.js',
                  {
                    method: 'usage-global',
                    absoluteImports: '/Users/uriklar/Code/honeybook-react/node_modules/core-js/index.js',
                    version: '3.8.1'
                  }
                ],
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-template-literals/lib/index.js'
              ]
            }
          }
        ],
        include: [ '/Users/uriklar/Code/honeybook-react' ],
        exclude: [ /node_modules/, /dist/ ]
      },
      {
        test: /\.js$/,
        use: [
          {
            loader: '/Users/uriklar/Code/honeybook-react/node_modules/babel-loader/lib/index.js',
            options: {
              sourceType: 'unambiguous',
              presets: [
                [
                  '/Users/uriklar/Code/honeybook-react/node_modules/@babel/preset-env/lib/index.js',
                  {
                    shippedProposals: true,
                    modules: false,
                    targets: 'defaults'
                  }
                ],
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/preset-react/lib/index.js'
              ],
              plugins: [
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-shorthand-properties/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-block-scoping/lib/index.js',
                [
                  '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-decorators/lib/index.js',
                  { legacy: true }
                ],
                [
                  '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-class-properties/lib/index.js',
                  { loose: true }
                ],
                [
                  '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-private-methods/lib/index.js',
                  { loose: true }
                ],
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-export-default-from/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-syntax-dynamic-import/lib/index.js',
                [
                  '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js',
                  { loose: true, useBuiltIns: true }
                ],
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-classes/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-arrow-functions/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-parameters/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-destructuring/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-spread/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-transform-for-of/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/babel-plugin-macros/dist/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-optional-chaining/lib/index.js',
                '/Users/uriklar/Code/honeybook-react/node_modules/@babel/plugin-proposal-nullish-coalescing-operator/lib/index.js',
                [
                  '/Users/uriklar/Code/honeybook-react/node_modules/babel-plugin-polyfill-corejs3/lib/index.js',
                  {
                    method: 'usage-global',
                    absoluteImports: '/Users/uriklar/Code/honeybook-react/node_modules/core-js/index.js',
                    version: '3.8.1'
                  }
                ]
              ]
            }
          }
        ],
        include: [Function: include]
      },
      {
        test: /\.css$/,
        use: [
          '/Users/uriklar/Code/honeybook-react/node_modules/style-loader/dist/cjs.js',
          {
            loader: '/Users/uriklar/Code/honeybook-react/node_modules/css-loader/dist/cjs.js',
            options: { importLoaders: 1 }
          }
        ]
      },
      {
        test: /\.(svg|ico|jpg|jpeg|png|apng|gif|eot|otf|webp|ttf|woff|woff2|cur|ani|pdf)(\?.*)?$/,
        loader: '/Users/uriklar/Code/honeybook-react/node_modules/file-loader/dist/cjs.js',
        query: { name: 'static/media/[name].[hash:8].[ext]' }
      },
      {
        test: /\.(mp4|webm|wav|mp3|m4a|aac|oga)(\?.*)?$/,
        loader: '/Users/uriklar/Code/honeybook-react/node_modules/url-loader/dist/cjs.js',
        query: { limit: 10000, name: 'static/media/[name].[hash:8].[ext]' }
      }
    ]
  },
  resolve: {
    extensions: [
      '.mjs', '.js',
      '.jsx', '.json',
      '.cjs', '.ts',
      '.tsx'
    ],
    modules: [ 'node_modules' ],
    mainFields: [ 'browser', 'main' ],
    alias: {
      '@emotion/core': '/Users/uriklar/Code/honeybook-react/node_modules/@emotion/core',
      '@emotion/styled': '/Users/uriklar/Code/honeybook-react/node_modules/@emotion/styled',
      'emotion-theming': '/Users/uriklar/Code/honeybook-react/node_modules/emotion-theming',
      '@storybook/addons': '/Users/uriklar/Code/honeybook-react/node_modules/@storybook/addons',
      '@storybook/api': '/Users/uriklar/Code/honeybook-react/node_modules/@storybook/api',
      '@storybook/channels': '/Users/uriklar/Code/honeybook-react/node_modules/@storybook/channels',
      '@storybook/components': '/Users/uriklar/Code/honeybook-react/node_modules/@storybook/components',
      '@storybook/core-events': '/Users/uriklar/Code/honeybook-react/node_modules/@storybook/core-events',
      '@storybook/router': '/Users/uriklar/Code/honeybook-react/node_modules/@storybook/router',
      '@storybook/theming': '/Users/uriklar/Code/honeybook-react/node_modules/@storybook/theming',
      '@storybook/ui': '/Users/uriklar/Code/honeybook-react/node_modules/@storybook/ui',
      react: '/Users/uriklar/Code/honeybook-react/node_modules/react',
      'react-dom': '/Users/uriklar/Code/honeybook-react/node_modules/react-dom'
    },
    plugins: [
      {
        apply: [Function: nothing],
        makePlugin: [Function],
        moduleLoader: [Function],
        topLevelLoader: { apply: [Function: nothing] },
        bind: [Function],
        tsLoaderOptions: [Function],
        forkTsCheckerOptions: [Function]
      }
    ]
  },
  resolveLoader: { plugins: [ { apply: [Function: nothing] } ] },
  recordsPath: '/Users/uriklar/Code/honeybook-react/node_modules/.cache/storybook/public/records.json',
  performance: { hints: false },
  optimization: {
    splitChunks: { chunks: 'all' },
    runtimeChunk: true,
    sideEffects: true,
    usedExports: true,
    concatenateModules: true,
    minimizer: []
  }
}
   ```

</details>


I tried these suggested solutions:
* Installing core-js and aliassing it in the webpack config
* Removing node_modules and lock file and re installing

Any suggestions?

**Comment by shilman** [comment]
@43081j any chance you can provide some support on this one? looks like people are still running into this issue even after your change.

**Comment by 43081j** [comment]
the previous problem with `__webpack_require__` feels like its another issue.

somehow the webpack overrides conflicting maybe, its difficult to reproduce that one for sure. @uriklar if you could throw together a test repo which reproduces that problem, that'd be a big help. a minimal reproduction, trying to strip away as much as you can without making the problem go away.

same for you @henriquez , if you can reproduce it in a repo somewhere we can try debug a bit

**Comment by hydrosquall** [comment]
Good news update on https://github.com/storybookjs/storybook/issues/11255#issuecomment-741905199 

I investigated this bug with @arcanis, who quickly identified the issue. We found the error was due to the fact that a dependency of the main application (the one using storybook) was in a separate yarn workspace that did not have `core-js` explicitly specified as a dependency. 

Adding the same version of `core-js` as a dependency to that library workspace as the version in the main application (the one where storybook is explicitly used) fixes the issue (using `Storybook 6.0.21`).

**Comment by Ayc0** [comment]
Also, I found another workaround: when using yarn pnp + workspaces, if `core-js` is available in the top level module but not in a sub-package, adding this line in the main.js file should work:

```js
config.resolve.alias["core-js"] = path.dirname(require.resolve("core-js"));
```

(`require.resolve("core-js")` will return the path to the `.../node-modules/core-js/index.js` file in pnp, and then `path.dirname(...)` will return the path to core-js itself and thus when `core-js/module` will be required, it'll indeed generate `.../node-modules/core-js/module` and not `.../node-modules/core-js/index.js/module`)

**Comment by obonyojimmy** [comment]
Simply install `core-js` dependecy `npm i core-js`

**Comment by jmlaya** [comment]
That workaround didn't work for me, I needed to configure specific aliases for core-js modules with conflicts:

`const path = require('path');

module.exports = {
  "stories": [
    "../src/**/*.stories.mdx",
    "../src/**/*.stories.@(js|jsx|ts|tsx)"
  ],
  "addons": [
    "@storybook/addon-links",
    "@storybook/addon-essentials"
  ],
  webpackFinal: async (config, { configType }) => {
    config.resolve.alias = {
      ...config.resolve.alias,
      'core-js/modules/es6.object.keys': path.resolve(
        __dirname,
        '../node_modules/core-js/modules/es.object.keys',
      ),
      'core-js/modules/es7.object.get-own-property-descriptors': path.resolve(
        __dirname,
        '../node_modules/core-js/modules/es.object.get-own-property-descriptors',
      ),
      'core-js/modules/web.dom.iterable': path.resolve(
        __dirname,
        '../node_modules/core-js/modules/web.dom-collections.iterator.js',
      ),
      'core-js/modules': path.resolve(
        __dirname,
        '../node_modules/core-js/modules',
      ),
    }

    return config;
  },
}
`

**Comment by Ayc0** [comment]
@jmlaya you should try the snippet I wrote:
```js
config.resolve.alias["core-js"] = path.dirname(require.resolve("core-js"));
```
I did it this way so that it'll work for all submodules si it should help you too

**Comment by charkour** [comment]
Adding `core-js` as a Dev Dep to the package I was importing solved my issue.

**Comment by jmlaya** [comment]
@Ayc0 I did something similar, but these files with these names does not exists anymore in the core-js package:

- es6.object.keys
- es7.object.get-own-property-descriptors
- web.dom.iterable

**Comment by chrisabrams** [comment]
> Also, I found another workaround: when using yarn pnp + workspaces, if `core-js` is available in the top level module but not in a sub-package, adding this line in the main.js file should work:
> 
> ```js
> config.resolve.alias["core-js"] = path.dirname(require.resolve("core-js"));
> ```
> 
> (`require.resolve("core-js")` will return the path to the `.../node-modules/core-js/index.js` file in pnp, and then `path.dirname(...)` will return the path to core-js itself and thus when `core-js/module` will be required, it'll indeed generate `.../node-modules/core-js/module` and not `.../node-modules/core-js/index.js/module`)

`dirname` was the magic I needed, thanks!

**Comment by henriquez** [comment]
> Simply install `core-js` dependecy `npm i core-js`

This worked for me when storybook 6 complained about core-js missing in dependency I was loading from another project, the key here is to install core-js *in that other project* not in the project that is using storybook.

**Comment by shaodowm** [comment]
> That workaround didn't work for me, I needed to configure specific aliases in the main.js file for core-js modules with name conflicts:
> 
> ```
> const path = require('path');
> 
> module.exports = {
>   "stories": [
>     "../src/**/*.stories.mdx",
>     "../src/**/*.stories.@(js|jsx|ts|tsx)"
>   ],
>   "addons": [
>     "@storybook/addon-links",
>     "@storybook/addon-essentials"
>   ],
>   webpackFinal: async (config, { configType }) => {
>     config.resolve.alias = {
>       ...config.resolve.alias,
>       'core-js/modules/es6.object.keys': path.resolve(
>         __dirname,
>         '../node_modules/core-js/modules/es.object.keys',
>       ),
>       'core-js/modules/es7.object.get-own-property-descriptors': path.resolve(
>         __dirname,
>         '../node_modules/core-js/modules/es.object.get-own-property-descriptors',
>       ),
>       'core-js/modules/web.dom.iterable': path.resolve(
>         __dirname,
>         '../node_modules/core-js/modules/web.dom-collections.iterator.js',
>       ),
>       'core-js/modules': path.resolve(
>         __dirname,
>         '../node_modules/core-js/modules',
>       ),
>     }
> 
>     return config;
>   },
> }
> ```

I have the same problem with you. You can try to add the option corejs: 3 with @babel/preset-env config in your .babelrc file and this works for me.
```
[
    "@babel/preset-env",
    {
      "modules": false,
      "targets": {
        "browsers": ["iOS >= 9"]
      },
      "useBuiltIns": "usage",
      "corejs": 3
    }
  ]
```
